### PR TITLE
Webpack upgrade

### PR DESCRIPTION
**Pull request by simonseyock**
This PR:
* integrates the munimap_transport python module into the munimap module
* Upgrades webpack from v4 to v5
* Upgrades node from v16 to v22
* Removes Flask-Assets
* Reworks structure of the frontend build: All frontend related files (except for templates) are now found in a `frontend` folder. Webpack then generates the entire `static` folder and Flask delivers it "as-is" without any modification
* Sass Files are now imported from JavaScript Files and the generated CSS files are hardwired into the templates (they are not likely to change much)

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,24 +1,21 @@
-ARG ANOL_COMMIT_HASH=08b57ecd563a1add5b4bdacdd026ffe4df603d20
+ARG ANOL_COMMIT_HASH=63de4627b24c5b886352858c2e52f60dcc3f16c0
 
-FROM node:14.19.3-alpine3.14 AS clientbuilder
+FROM node:22-alpine@sha256:6e80991f69cc7722c561e5d14d5e72ab47c0d6b6cfb3ae50fb9cf9a7b30fdf97 AS clientbuilder
```
Comment (jansule) [inline Dockerfile:3]: Is there a reason for using a specific hash here?

File: Dockerfile [hunk 2]
```diff
@@ -1,24 +1,21 @@
...
-RUN npm i -g npm@7
-
 RUN mkdir -p tmp/anol
 RUN cd /tmp/anol \
     && wget https://github.com/terrestris/anol/archive/$ANOL_COMMIT_HASH.zip -O anol.zip \
     && unzip anol.zip \
     && mv anol-$ANOL_COMMIT_HASH /anol
 
-RUN cd /anol && npm ci
+RUN cd /anol && npm ci --omit=dev
```
Comment (jansule) [inline Dockerfile:14]: Don't we need the dev dependencies for building the frontend?

File: dev/README.md [hunk 3]
```diff
@@ -35,7 +35,7 @@ wget -q -O- https://repo1.maven.org/maven2/org/mapfish/print/print-cli/3.9.0/pri
 cd ../../
 git clone git@github.com:terrestris/anol.git
 cd anol/
-npm i
+npm i --omit=dev
```
Comment (jansule) [inline dev/README.md:38]:
```suggestion
npm ci --omit=dev
```

File: munimap/frontend/sass/app.sass [hunk 4]
```diff
@@ -308,3 +320,212 @@
   .tooltip + .btn,
   .tooltip + .btn-group
     margin-left: -1px
+
+
+// TODO: is this also needed in admin?
```
Comment (jansule) [inline munimap/frontend/sass/app.sass:325]: Is this TODO still relevant?

File: munimap/templates/munimap/app/iframe.html [hunk 5]
```diff
@@ -6,16 +6,10 @@
...
-        <link href="{$ ASSET_URL $}" rel="stylesheet" type="text/css">
-      {% endassets %}
-      {% assets "app-css" %}
-        <link href="{$ ASSET_URL $}" rel="stylesheet" type="text/css">
-      {% endassets %}
-      {% assets "draw-libs-css" %}
-        <link href="{$ ASSET_URL $}" rel="stylesheet" type="text/css">
-      {% endassets %}
-    {% endblock %}
+      <!-- TODO: libs-css -->
```
Comment (jansule) [inline munimap/templates/munimap/app/iframe.html:9]: Are these TODOs still relevant?

File: munimap/templates/munimap/base.html [hunk 6]
```diff
@@ -5,16 +5,17 @@
       {% include 'munimap/meta.html' %}
     {% endblock %}
 
+    <!-- TODO: integrate as imports into JS Files -->
```
Comment (jansule) [inline munimap/templates/munimap/base.html:8]: Is this TODO still relevant?

File: munimap/templates/munimap/pages/base.html [hunk 7]
```diff
@@ -1,5 +1,10 @@
 {% extends "munimap/base.html" %}
 
+{% block header %}
+<!-- TODO: we could generate specific css just for pages -->
```
Comment (jansule) [inline munimap/templates/munimap/pages/base.html:4]: Is this TODO still relevant?

**Review by jansule — COMMENTED** [review]
Nice Work!

File: Dockerfile (hunk 2 above)
Comment (simonseyock) [reply Dockerfile:14]: No, the anol dev dependencies are not needed. Anol is included via source files and is built by webpack in the bielfeldGEOCLIENT.

File: munimap/templates/munimap/app/iframe.html (hunk 5 above)
Comment (simonseyock) [reply munimap/templates/munimap/app/iframe.html:9]: Yes, good catch!

File: Dockerfile (hunk 1 above)
Comment (simonseyock) [reply Dockerfile:3]: No, this is just for pinning the image. We have renovate installed which should inform us about newer hashes.

**Pull request merged**